OpenRA.Mods.Common/Widgets/SupportPowersWidget.cs

```diff
@@ -125,18 +129,59 @@ public class SupportPowerIcon
 			public HotkeyReference Hotkey;
 		}
 
+		bool IsPowersModified()
+		{
+			var modified = false;
+			if (powersModified)
+			{
+				cachedPowers = spm.Powers.Values.OrderBy(p => p.Info.SupportPowerPaletteOrder).ToArray();
+				powersModified = false;
+				modified = true;
+			}
+
+			if (!modified)
+			{
+				var disabledCount = 0;
+				foreach (var p in cachedPowers)
+				{
+					// There are few Powers and even less Disabled
+					if (p.Disabled)
```

Comment:
Can we please have the `SupportPowerManager` fire the `OnPowersChanged` so we don't need to check this here?

Reply:
> Can we please have the `SupportPowerManager` fire the `OnPowersChanged` so we don't need to check this here?

You mean to say .. let the manager also notice that powers were disabled? This is not possible, because then the manager would need to poll the powers? And/or the Powers would need to fire a Disabled state change event?

Reply:
I see.

Looking at this in more detail, it seems that `SupportPowerInstance.Disabled` should only be able to change (a) in response to the power being activated (b) in response to the `SupportPowerManager` ticking it (c) in response to conditions changing (which i *think* should be fine being rolled into (b)?).

So doesn't "the manager would need to poll the powers" just amount to checking the state before and after using the instance in `SupportPowerManager.ResolveOrder` and `SupportPowerManager.Tick`?

I may be missing something here, however.

Reply:
I did not look into from where `Disabled` is modified. Assuming that in future any code could set this field? But if you believe it can only originate from an order - then that perhaps could work.

I am fine with closing this issue.

Reply:
I don't think we should close this - quite the opposite, I think this isn't going far enough!